sql/opt_hints.h

```diff
@@ -492,6 +507,21 @@ class Opt_hints_qb : public Opt_hints
 
   virtual void append_name(THD *thd, String *str) override
   {
+    /*
+      If an explicit name isn't given, then the implicit
+      query block name (e.g. `select#X`) will be used.  But
+      we don't yet support that for VIEWs, so don't show it.
+    */
+    if ((thd->lex->sql_command == SQLCOM_CREATE_VIEW ||
+         thd->lex->sql_command == SQLCOM_SHOW_CREATE) &&
+        is_print_name_default())
+      return;
+
+    if (thd->lex->derived_tables & DERIVED_VIEW &&
+        thd->lex->sql_command == SQLCOM_SELECT &&
+        is_print_name_default())
```

Comment:
This looks a bit odd. 
Why check for SQLCOM_SELECT? What about multi-table UPDATE or DELETE or SQLCOM_INSERT_SELECT?  Their processing is very similar to that of SELECT so it looks very surprising that there is a difference in how hint name would be printed.